api/src/OpenTelemetry/Internal/Logs/Types.hs

```diff
@@ -58,7 +58,7 @@ data LogRecordExporter body = LogRecordExporter
   -- Result:
   -- Success - The batch has been successfully exported. For protocol exporters this typically means that the data is sent over the wire and delivered to the destination server.
   -- Failure - exporting failed. The batch must be dropped. For example, this can happen when the batch contains bad data and cannot be serialized.
-  , logRecordExporterForceFlush :: IO ()
+  , logRecordExporterForceFlush :: IO FlushResult
```

Comment:
It may not be necessary for `forceFlush` to return a `FlushResult` at every step in the chain of `Exporter` to `Processor` to `LoggerProvider` but I think it makes sense for each step to be able to communicate a failure or timeout.